bool bottom
bool front
---
bool out
